cli/main.go

```diff
@@ -29,155 +26,44 @@ func fail(err string) {
 	os.Exit(1)
 }
 
-// ExtraRule configures a new rule to be used with revive.
-type ExtraRule struct {
-	Rule          lint.Rule
-	DefaultConfig lint.RuleConfig
-}
-
-// NewExtraRule returns a configured extra rule
-func NewExtraRule(rule lint.Rule, defaultConfig lint.RuleConfig) ExtraRule {
-	return ExtraRule{
-		Rule:          rule,
-		DefaultConfig: defaultConfig,
-	}
-}
-
 // RunRevive runs the CLI for revive.
-func RunRevive(extraRules ...ExtraRule) {
-	log, err := logging.GetLogger()
-	if err != nil {
-		fail(err.Error())
-	}
-
-	formatter, err := config.GetFormatter(formatterName)
-	if err != nil {
-		fail(err.Error())
-	}
-
+func RunRevive(extraRules ...revivelib.ExtraRule) {
 	conf, err := config.GetConfig(configPath)
 	if err != nil {
 		fail(err.Error())
 	}
 
-	if setExitStatus {
-		conf.ErrorCode = 1
-		conf.WarningCode = 1
-	}
-
-	extraRuleInstances := make([]lint.Rule, len(extraRules))
-	for i, extraRule := range extraRules {
-		extraRuleInstances[i] = extraRule.Rule
-
-		ruleName := extraRule.Rule.Name()
-		_, isRuleAlreadyConfigured := conf.Rules[ruleName]
-		if !isRuleAlreadyConfigured {
-			conf.Rules[ruleName] = extraRule.DefaultConfig
-		}
-	}
-
-	lintingRules, err := config.GetLintingRules(conf, extraRuleInstances)
+	revive, err := revivelib.New(
+		conf,
+		setExitStatus,
+		maxOpenFiles,
+		excludePaths,
+		extraRules...,
+	)
 	if err != nil {
 		fail(err.Error())
 	}
 
-	log.Println("Config loaded")
-
-	if len(excludePaths) == 0 { // if no excludes were set in the command line
-		excludePaths = conf.Exclude // use those from the configuration
-	}
-
-	packages, err := getPackages(excludePaths)
+	failures, err := revive.Lint(flag.Args()...)
```

Comment:
here I'm not convinced with only passing `includePatterns` and not also `excludePatterns`.
my feeling is both should be handled in the same way: either they are both a property of the `revive` lib instance or both are parameters of the `Lint` method.
I prefer to pass them as arguments to the `Lint` methods because it will allow a flexible usage patter: create a `revive` instance and then call `Lint` many times with different  include & exclude patterns. 

What  do you think?

Reply:
I was thinking more on the lines of "Lint these packages". The excludes part applies to all of them. But I can see your point and have no strong feelings. Will change!